src/protocol/step.rs

```diff
@@ -74,9 +74,14 @@ impl<S: Step + ?Sized> StepNarrow<S> for Descriptive {
             assert!(!s.contains('/'), "The string for a step cannot contain '/'");
         }
 
-        Self {
-            id: self.id.clone() + "/" + step.as_ref(),
+        let mut id = self.id.clone() + "/";
+        #[cfg(feature = "step-trace")]
+        {
+            id += &(std::any::type_name::<S>().to_owned() + "::");
```

Comment:
This will create more garbage than necessary because it creates an owned copy of the type name and an owned copy of the type name plus "::".  I don't see why the following wouldn't work:

```suggestion
            id += std::any::type_name::<S>() + "::";
```

Reply:
Since the larger goal here is to eliminate use of strings for step representation in performance-sensitive contexts, it's not worth a lot of fiddling, but another possibility is to use [concat](https://doc.rust-lang.org/std/primitive.slice.html#method.concat): `[std::any::type_name::<S>(), "::"].concat()`.

Reply:
Or even:

```rust
id += std::any::type_name::<S>();
id += "::";
```

Though Andy's point holds :)